tools/workspace-plugin/src/generators/prepare-initial-release/lib/utils.ts

```diff
@@ -0,0 +1,43 @@
+import type { Tree } from '@nx/devkit';
+import ignore from 'ignore';
+import { join, relative, sep } from 'node:path';
+
+/**
+ * Utility to act on all files in a tree that are not ignored by git.
+ * copied from https://github.com/nrwl/nx/blob/master/packages/devkit/src/generators/visit-not-ignored-files.ts
+ * utility override from original in order to process files that are ignored within .nxignore
+ */
+export function visitNotGitIgnoredFiles(tree: Tree, dirPath = tree.root, visitor: (path: string) => void): void {
+  // TODO (v17): use packages/nx/src/utils/ignore.ts
+  let ig = ignore();
+  if (tree.exists('.gitignore')) {
+    // ig = ignore();
+    ig.add('.git');
+    ig.add(tree.read('.gitignore', 'utf-8')!);
+  }
+  // if (tree.exists('.nxignore')) {
```

Comment:
i should probably remove this, but for sake of copy-paste diff maybe keep it ? 🤷